tests/val.rs

```diff
@@ -26,22 +26,35 @@ fn validate_tool(_input: &[u8]) -> Option<Result<(), spirv_tools::Error>> {
 
 #[test]
 fn gets_error_message() {
-    let cexpected_msg = "error:0:0 - Loop header '6[%loop_header]' is targeted by 2 back-edge blocks but the standard requires exactly one\n  %loop_header = OpLabel\n";
-    let texpected_msg = "error:0:0 - Loop header '6[%loop_header]' is targeted by 2 back-edge blocks but the standard requires exactly one";
+    let expected_msg = "error:0:0 - Loop header '6[%loop_header]' is targeted by 2 back-edge blocks but the standard requires exactly one";
+    let expected_notes = "  %loop_header = OpLabel";
+
     match (validate_compiled(SPIRV_BIN), validate_tool(SPIRV_BIN)) {
         (Some(resc), Some(rest)) => {
             let cstr = resc.unwrap_err().to_string();
             let tstr = rest.unwrap_err().to_string();
-            assert_eq!(&cstr[..111], &tstr[..111]);
 
-            assert_eq!(cstr, cexpected_msg);
-            assert_eq!(tstr, texpected_msg);
+            assert_eq!(cstr, tstr);
+
+            assert_eq!(&tstr[..113], expected_msg);
+            assert_eq!(&cstr[..113], expected_msg);
+
+            assert_eq!(&cstr[113 + 1..], expected_notes);
+            assert_eq!(&tstr[113 + 1..], expected_notes);
         }
         (Some(resc), None) => {
-            assert_eq!(resc.unwrap_err().to_string(), cexpected_msg);
+            let diag = resc.unwrap_err().diagnostic.unwrap();
+            assert_eq!(diag.line, 0);
+            assert_eq!(diag.column, 0);
+            assert_eq!(diag.message, &expected_msg[12..]);
+            assert_eq!(diag.notes, expected_notes);
         }
         (None, Some(rest)) => {
-            assert_eq!(rest.unwrap_err().to_string(), texpected_msg);
+            let diag = rest.unwrap_err().diagnostic.unwrap();
+            assert_eq!(diag.line, 0);
+            assert_eq!(diag.column, 0);
+            assert_eq!(diag.message, &expected_msg[12..]);
+            assert_eq!(diag.notes, expected_notes);
         }
         _ => {}
```

Comment:
Surely this could be iterating over both approaches and processing them the same?
At most `assert_eq!(cstr, tstr);` could be done as an *additional* check, when both are available, but the rest should be uniform I think.